src/Apps/Artwork/Components/ArtworkDetails/RequestConditionReport.tsx

```diff
@@ -0,0 +1,205 @@
+import { Button, Col, Modal, Row, Sans, Serif, Spinner } from "@artsy/palette"
+import React, { useState } from "react"
+import { commitMutation, graphql, RelayProp } from "react-relay"
+
+import { AnalyticsSchema, track, useSystemContext, useTracking } from "Artsy"
+import * as Schema from "Artsy/Analytics/Schema"
+import { ErrorModal } from "Components/Modal/ErrorModal"
+import * as authentication from "Components/NavBar/Utils/authentication"
+import createLogger from "Utils/logger"
+
+import { RequestConditionReportMutation } from "__generated__/RequestConditionReportMutation.graphql"
+
+const logger = createLogger(
+  "Apps/Artwork/Components/ArtworkDetails/RequestConditionReport"
+)
+
+export interface RequestConditionReportProps {
+  saleArtworkID: string
+  relay?: RelayProp
+}
+
+export const RequestConditionReport: React.FC<RequestConditionReportProps> = track(
+  {
+    context_module: AnalyticsSchema.ContextModule.RequestConditionReport,
+  }
+)(props => {
+  const { mediator, user } = useSystemContext()
+  const { trackEvent } = useTracking()
+
+  const [requestingConditionReport, setRequestingConditionReport] = useState(
+    false
+  )
+
+  const [
+    showConditionReportRequestedModal,
+    setShowConditionReportRequestedModal,
+  ] = useState(false)
+
+  const [showErrorModal, setShowErrorModal] = useState(false)
+
+  const isLoggedIn = Boolean(user)
+  const { saleArtworkID, relay } = props
+
+  const requestConditionReport = () => {
+    return new Promise(async (resolve, reject) => {
+      commitMutation<RequestConditionReportMutation>(relay.environment, {
+        onCompleted: data => {
+          resolve(data)
+        },
+        onError: error => {
+          reject(error)
+        },
+        mutation: graphql`
+          mutation RequestConditionReportMutation(
+            $input: RequestConditionReportInput!
+          ) {
+            requestConditionReport(input: $input) {
+              conditionReportRequest {
+                internalID
+              }
+            }
+          }
+        `,
+        variables: {
+          input: { saleArtworkID },
+        },
+      })
+    })
+  }
+
+  const handleMutationError = error => {
+    logger.error(error)
+
+    let errorMessages: string[]
```

Comment:
This stanza is the same as the one in the [Register app](https://github.com/artsy/reaction/blob/master/src/Apps/Auction/Routes/Register/index.tsx#L156-L163).

Have you considered abstracting this a bit? Seems like this could be a function named something like `standardizeError(error: Error): string[]`. Just wanted to surface the idea!

Reply:
The shared bit is:

```typescript
    if (Array.isArray(error)) {
      errorMessages = error.map(e => e.message)
    } else if (typeof error === "string") {
      errorMessages = [error]
    } else if (error.message) {
      errorMessages = [error.message]
    }
```

We actually don't need most of it. We're not dealing with multiple errors and we're not passing strings as errors either. I'll go through and simplify this.